src/request/providers/node.ts

```
@@ -0,0 +1,558 @@
+								}
+								else if (encoding === 'gzip') {
+									zlib.gunzip(dataAsBuffer, function (err: Error, result: Buffer) {
+										if (err) {
+											reject(err);
+										}
+
+										dataAsBuffer = result;
+										handleEncoding();
+									});
+								}
+								else if (encoding === 'deflate') {
+									zlib.inflate(dataAsBuffer, function (err: Error, result: Buffer) {
+										if (err) {
+											reject(err);
+										}
+
+										dataAsBuffer = result;
+										handleEncoding();
+									});
+								}
+								else {
+									reject(new Error('Unsupported content encoding, ' + encoding));
+								}
+							}
+							else {
+								data.data = String(dataAsBuffer);
+
+								response.emit({
+									type: 'end',
+									response
+								});
+
+								resolve(message);
+							}
+						};
+
+						handleEncoding();
+					});
+				});
+			}, () => {
+				request.abort();
+			});
+
+			const data: RequestData = {
+				task,
+				buffer: [],
+				data: '',
+				size: 0,
+				used: false,
+				url: url,
+				requestOptions: options,
+				nativeResponse: message
+			};
+
+			dataMap.set(response, data);
+
+			resolve(response);
+		});
+
+		request.once('error', reject);
+
+		if (options.body) {
+			if (options.body instanceof Buffer) {
+				request.end(options.body.toString());
+			}
+			else {
+				request.end(options.body.toString());
+			}
+		}
+		else {
+			request.end();
+		}
+
+		if (options.timeout > 0 && options.timeout !== Infinity) {
+			timeoutHandle = createTimer(() => {
+				timeoutReject && timeoutReject(new TimeoutError('The request timed out'));
+			}, options.timeout);
+		}
+	}, () => {
+		request.abort();
+	}).catch(function (error: Error): any {
+		const parsedUrl = urlUtil.parse(url);
+
+		if (parsedUrl.auth) {
+			parsedUrl.auth = '(redacted)';
+		}
+
+		const sanitizedUrl = urlUtil.format(parsedUrl);
+
+		error.message = '[' + requestOptions.method + ' ' + sanitizedUrl + '] ' + error.message;
```

Are there cases where setting `error.message` would throw?

---

Maybe, but either way we're throwing an error. I could swallow errors to `error.message` and just ignore it if it didn't work?  What do you think?

---

I'm really not sure what the best practice is here. At this point, it might be best to just leave this and if we hit this in reality (rather than trying to cover every potential case), we can adjust.

---

OK. Let's ignore this for now.